testing/runner.ts

```diff
@@ -86,36 +86,42 @@ export class Runner {
         if (result.err) {
           const errString = result.err.stack
             ? result.err.stack && indent(result.err.stack as string)
-            : `    ${JSON.stringify(result.err, null, 4)}`;
+            : `    ${DeterministicStringify(result.err, { space: "  " })}`;
 
           console.error(`\n${errString}\n`);
           if (result.err.showDiff) {
             if (result.err.expected) {
               console.error(`    Expected:\n`);
-              console.error(indent(JSON.stringify(result.err.expected, null, 4), 8));
+              console.error(indent(DeterministicStringify(result.err.expected, { space: "  " })));
             }
             if (result.err.actual) {
               console.error(`\n    Actual:\n`);
-              console.error(indent(JSON.stringify(result.err.actual, null, 4), 8));
+              console.error(indent(DeterministicStringify(result.err.actual, { space: "  " })));
             }
             const green = "\x1b[32m";
             const red = "\x1b[31m";
             const reset = "\x1b[0m";
             if (result.err.actual && result.err.expected) {
-              console.error(
-                `\n    Overall diff (${green}expected${reset}, ${red}actual${reset}):\n`
-              );
               const diffs = Diff.diffJson(
                 DeterministicStringify(result.err.expected, { space: "  " }),
                 DeterministicStringify(result.err.actual, { space: "  " })
               );
-              let toLog = "";
-              diffs.forEach(diff => {
-                // This diff won't show well for users with either default green or red text.
-                const colorPrefix = diff.added ? red : diff.removed ? green : reset;
-                toLog += `${colorPrefix}${diff.value}`;
-              });
-              console.error(toLog);
+              if (diffs.length === 1 && !diffs[0].added && !diffs[0].removed) {
+                console.error(
+                  `\n    ${green}JSONs appear identical! Have all protobufs been casted?`
```

Comment:
I'd like this to not be too specific to us, can you put a more generic message, e.g. "Are you comparing objects with functions?"

Reply:
Sure

Reply:
Sure